### DISCUSSION TITLE
Implement remaining material allocation

### DISCUSSION THREAD

**Discussion by maxware0715** [frePPLe/frepple#621] [Q&A] [answered]
Can production planning schedule with "remaining material allocation" method?
For example, I have input items with different priorities. 
Before create new purchase order of item with highest priority, existed items with any priorities should be consumed.

**Comment by jdetaeye** [comment]
I think your question relates to a feature which we call "alternate materials".   See https://frepple.com/docs/current/examples/buffer/alternate-materials.php 
 
Rephrasing your question, it would be like:
I can make my end product using raw material A (preferred) OR raw material B (second choice) OR raw material C (least preferred).  Before buying more of raw material A, I want to use up all my existing supply of A, B and C.
This is the second option described in the above link.

**Reply by maxware0715** [reply]
Thank you very much for rephrasing the question.

But my case would be more complicated.
For example:
operation materials A: 
    start: A1 and A2
    end: A
operation materials B: 
    start: B1 and B2
    end: B
operation materials C: 
    start: A or B, prioriy: A:1, B:2

Order: C

Initially, I set up confirmed manufacturing orders which produce A1,B1,
expect new purchase order of A2,B2 will be created to consumed existed items.

However, planning engine only create purchase order for A and ignore existed B1.

How can I set up to consume both A1 and B1?

**Reply by jdetaeye** [reply, marked as answer]
I see....   Your confirmed supply is deeper in the bill of materials, and the configuration to let the algorithm scan & use it is a bit more complicated.

You'll need to switch the "search mode" of the operations from the default "priority" to "minimum cost". And you will need to set a cost on items, resources and/or operations. The algorithm will then evaluate each alternates A & B, and select the cheapest one.  Since using up the existing supply/inventory is evaluated to be cheaper compared to proposing new production or purchases, you'll get the results you want. 
In summary: it's (unfortunately) not that simple to get it all set up correctly.
